client/state/imports/actions.js

```diff
@@ -55,38 +54,39 @@ const expiryOrder = ( siteId, importerId ) =>
 const importOrder = importerStatus =>
 	toApi( Object.assign( {}, importerStatus, { importerState: appStates.IMPORTING } ) );
 
-const apiStart = () => Dispatcher.handleViewAction( { type: IMPORTS_FETCH } );
-const apiSuccess = data => {
-	Dispatcher.handleViewAction( { type: IMPORTS_FETCH_COMPLETED } );
+const apiStart = () => ( { type: IMPORTS_FETCH } );
+
+const apiSuccess = dispatch => data => {
+	dispatch( { type: IMPORTS_FETCH_COMPLETED } );
 
 	return data;
 };
-const apiFailure = data => {
-	Dispatcher.handleViewAction( { type: IMPORTS_FETCH_FAILED } );
+const apiFailure = dispatch => data => {
+	dispatch( { type: IMPORTS_FETCH_FAILED } );
 
 	return data;
 };
 const setImportLock = ( shouldEnableLock, importerId ) => {
 	const type = shouldEnableLock ? IMPORTS_IMPORT_LOCK : IMPORTS_IMPORT_UNLOCK;
 
-	Dispatcher.handleViewAction( { type, importerId } );
+	return { type, importerId };
 };
 const lockImport = partial( setImportLock, true );
 const unlockImport = partial( setImportLock, false );
```

Comment:
If these two functions were just
```js
const lockImport = importerId => ( { type: IMPORTS_IMPORT_LOCK, importerId } );
const unlockImport = importerId => ( { type: IMPORTS_IMPORT_UNLOCK, importerId } );
```
they would be admittedly more boring, but maybe more readable and shorter.